bindings/node/src/mongocrypt.cc

```diff
@@ -0,0 +1,689 @@
+        }
+
+        mongocrypt_binary_t* binary = BufferToBinary(keyId);
+        if (!mongocrypt_ctx_setopt_key_id(context, binary)) {
+            return std::make_pair(false, errorStringFromStatus(context));
+        }
+    }
+
+    if (Nan::Has(options, ALGORITHM_KEY).FromMaybe(false)) {
+        std::string algorithm = StringOptionValue(options, "algorithm");
+        if (!mongocrypt_ctx_setopt_algorithm(
+                context, const_cast<char*>(algorithm.c_str()), algorithm.size())) {
+            return std::make_pair(false, errorStringFromStatus(context));
+        }
+    }
+
+    if (Nan::Has(options, INITIALIZATION_VECTOR_KEY).FromMaybe(false)) {
+        if (!Nan::Get(options, INITIALIZATION_VECTOR_KEY).ToLocalChecked()->IsObject()) {
+            return std::make_pair(false, "`initializationVector` must be a Buffer");
+        }
+
+        v8::Local<v8::Object> initializationVector =
+            Nan::To<v8::Object>(Nan::Get(options, INITIALIZATION_VECTOR_KEY).ToLocalChecked())
+                .ToLocalChecked();
+        if (!node::Buffer::HasInstance(initializationVector)) {
+            return std::make_pair(false, "`initializationVector` must be a Buffer");
+        }
+
+        mongocrypt_binary_t* binary = BufferToBinary(initializationVector);
+        if (!mongocrypt_ctx_setopt_initialization_vector(context, binary)) {
+            return std::make_pair(false, errorStringFromStatus(context));
+        }
+    }
+
+    return std::make_pair(true, std::string());
+}
+
+Nan::Persistent<v8::Function> MongoCrypt::constructor;
+NAN_MODULE_INIT(MongoCrypt::Init) {
+    v8::Local<v8::FunctionTemplate> tpl = Nan::New<v8::FunctionTemplate>(New);
+    tpl->SetClassName(Nan::New("MongoCrypt").ToLocalChecked());
+    tpl->InstanceTemplate()->SetInternalFieldCount(1);
+
+    Nan::SetPrototypeMethod(tpl, "makeEncryptionContext", MakeEncryptionContext);
+    Nan::SetPrototypeMethod(tpl, "makeExplicitEncryptionContext", MakeExplicitEncryptionContext);
+    Nan::SetPrototypeMethod(tpl, "makeDecryptionContext", MakeDecryptionContext);
+    Nan::SetPrototypeMethod(tpl, "makeExplicitDecryptionContext", MakeExplicitDecryptionContext);
+    Nan::SetPrototypeMethod(tpl, "makeDataKeyContext", MakeDataKeyContext);
+
+    v8::Local<v8::ObjectTemplate> itpl = tpl->InstanceTemplate();
+    itpl->SetInternalFieldCount(1);
+
+    Nan::SetAccessor(itpl, Nan::New("status").ToLocalChecked(), Status);
+
+    constructor.Reset(Nan::GetFunction(tpl).ToLocalChecked());
+    Nan::Set(
+        target, Nan::New("MongoCrypt").ToLocalChecked(), Nan::GetFunction(tpl).ToLocalChecked());
+}
+
+void MongoCrypt::logHandler(mongocrypt_log_level_t level,
+                            const char* message,
+                            uint32_t message_len,
+                            void* ctx) {
+    MongoCrypt* mongoCrypt = static_cast<MongoCrypt*>(ctx);
+    if (!mongoCrypt) {
+        fprintf(stderr, "Log handler called without `MongoCrypt` instance\n");
```

Comment:
q: why are these going directly to stderr and not being propagated back up to the js side?